hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -1518,6 +1521,25 @@ public int executeUpdate(String query, QueryParameters queryParameters) throws H
 		return result;
 	}
 
+	private void preventImmutableEntityUpdate(HQLQueryPlan plan) {
+		if ( plan.isUpdate() ) {
+			for ( EntityPersister entityPersister : getSessionFactory().getMetamodel().entityPersisters().values() ) {
+				if ( !entityPersister.isMutable() ) {
+					List<Serializable> entityQuerySpaces = new ArrayList<>(
+							Arrays.asList( entityPersister.getQuerySpaces() )
+					);
+					entityQuerySpaces.retainAll( plan.getQuerySpaces() );
+
+					if ( !entityQuerySpaces.isEmpty() ) {
```

Comment:
See my comment on the Jira about log versus exception.  I's like to have a consensus on that.  Other than the impact on this specific line, I think this looks good